authentication/account/repository/identity.go

```diff
@@ -415,12 +415,13 @@ func (m *GormIdentityRepository) ListIdentitiesToNotifyForDeactivation(ctx conte
 // and for whom there is a `deactivation_notification` value.
 // The result size is limited to the given number of identities (ordered by last activity)
 // if limit is a negative value (eg: '-1'), it is ignored
-func (m *GormIdentityRepository) ListIdentitiesToDeactivate(ctx context.Context, lastActivity time.Time, limit int) ([]Identity, error) {
+func (m *GormIdentityRepository) ListIdentitiesToDeactivate(ctx context.Context, lastActivity, notification time.Time, limit int) ([]Identity, error) {
 	defer goa.MeasureSince([]string{"goa", "db", "user", "listIdentitiesToDeactivate"}, time.Now())
 	var identities []Identity
 	// sort identities by most inactive and then by date of creation to make sure we always get the same sublist of identities between
 	// queries to notify before deactivation and queries to deactivate for real.
-	err := m.db.Model(&Identity{}).Where("last_active < ? and deactivation_notification is NOT NULL", lastActivity).Order("last_active, created_at").Limit(limit).Find(&identities).Error
+	// Also, make sure that the users to deactivate were notified at least `n` days ealier.
+	err := m.db.Model(&Identity{}).Where("last_active < ? and deactivation_notification < ?", lastActivity, notification).Order("last_active, created_at").Limit(limit).Find(&identities).Error
```

Comment:
yes, that user's case covers it: https://github.com/xcoulon/fabric8-auth/blob/IssueODC394_deactivate_n_days_after_notification/authentication/account/service/user_blackbox_test.go#L233 -> the user has been inactive for a long-enough period, but since she did not get the notification, she is not listed in the users to deactivate.